modules/basicfuncs/list-funcs.c

```diff
@@ -151,6 +164,8 @@ _list_nth(gint argc, GString *argv[], GString *result, gint ndx)
   list_scanner_init(&scanner);
   list_scanner_input_gstring_array(&scanner, argc, argv);
 
+  _adjust_list_index(argc, argv, &ndx);
```

Comment:
This is clearly just a personal opinion! Leave it as is if you want.
I would prefer: _translate_negative_index_into_nth